src/components/IdealImage/index.js

```diff
@@ -158,6 +157,8 @@ export default class IdealImage extends Component {
         lqip: PropTypes.string.isRequired,
       }),
     ]).isRequired,
+    /** react component to determine visibility */
+    VisibilityObserver: PropTypes.node.isRequired,
```

Comment:
not sure about this one. Need to check if it is actually works

Reply:
yeah this is still a WIP